### PR TITLE
kola/kubeadm: test various CNIs

### PR DESCRIPTION
**Pull request by tormath1**
In this PR, we try to cover more kubernetes use cases by testing different CNIs.

- flannel
- calico
- cilium

## How to use

```
$ ./bin/kola list | grep kubeadm
kubeadm.calico.base			[aws azure do gce openstack packet qemu qemu-unpriv]	[all]		[cl]		[all]			[all]
kubeadm.cilium.base			[aws azure do gce openstack packet qemu qemu-unpriv]	[all]		[cl]		[all]			[all]
kubeadm.flannel.base			[aws azure do gce openstack packet qemu qemu-unpriv]	[all]		[cl]		[all]			[all]
$ sudo ./bin/kola run -b cl -p qemu --qemu-image ./flatcar_production_qemu_image.img kubeadm.flannel.base
```

closes: https://github.com/kinvolk/Flatcar/issues/309

### CODE REVIEW

File: kola/tests/kubeadm/templates.go [hunk 1]
```diff
@@ -307,8 +317,23 @@ EOF
     cp /etc/kubernetes/admin.conf /home/core/.kube/config
     chown -R core:core /home/core/.kube; chmod a+r /home/core/.kube/config;
 
+{{ if eq .CNI "calico" }}
     kubectl create -f https://docs.projectcalico.org/manifests/tigera-operator.yaml
     kubectl apply -f calico.yaml
+{{ end }}
+{{ if eq .CNI "flannel" }}
+    curl -sSL https://raw.githubusercontent.com/flannel-io/flannel/master/Documentation/kube-flannel.yml > kube-flannel.yml
```
Comment (pothos) [inline kola/tests/kubeadm/templates.go:325]:
```suggestion
    curl -sSfL https://raw.githubusercontent.com/flannel-io/flannel/master/Documentation/kube-flannel.yml > kube-flannel.yml
```
We should fail on 404

File: kola/tests/kubeadm/testdata/master-flannel-script.sh [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    systemctl enable --quiet --now kubelet
+    kubeadm config images pull
+    kubeadm init --config kubeadm-config.yaml
+    cp /etc/kubernetes/admin.conf $HOME/.kube/config
+    cp /etc/kubernetes/admin.conf /home/core/.kube/config
+    chown -R core:core /home/core/.kube; chmod a+r /home/core/.kube/config;
+
+
+
+    curl -sSL https://raw.githubusercontent.com/flannel-io/flannel/master/Documentation/kube-flannel.yml > kube-flannel.yml
```
Comment (pothos) [inline kola/tests/kubeadm/testdata/master-flannel-script.sh:75]: same

**Review by pothos — COMMENTED** [review]
Two small issues with fetching via curl but otherwise feel free to merge

**Comment by tormath1** [comment]
thanks @pothos for reviewing. Comments have been addressed. FYI `kola/tests/kubeadm/testdata/master-flannel-script.sh` is just a testdata to assert master script is well generated from the template.

Once the CI :green_circle: I'll rebase and merge.

**Review by pothos — APPROVED** [review]

**Pull request merged**